src/Handler/Command.hs

```diff
@@ -64,9 +70,14 @@ getCommandR token = do
             $(widgetFile "command")
 
 deleteCommandR :: Token -> Handler ()
-deleteCommandR token = unsafeRunStorage $ do
-    del token
-    del $ History token
+deleteCommandR token = do
+    archived <- unsafeRunStorage $ do
```

Comment:
Looks like it's an instance of `MonadThrow`, which I think would work.

http://haddock.stackage.org/lts-3.7/yesod-core-1.4.15/Yesod-Core-Types.html#v:HandlerT